src/Features/Core/Portable/ChangeAccessibilityModifier/AbstractChangeAccessibilityModifierCodeFixProvider.cs

```diff
@@ -0,0 +1,116 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable disable
+
+using System;
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.LanguageServices;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+
+namespace Microsoft.CodeAnalysis.ChangeAccessibilityModifier
+{
+    internal abstract class AbstractChangeAccessibilityModifierCodeFixProvider : CodeFixProvider
+    {
+        private readonly string _title = FeaturesResources.Change_accessibility;
+        private readonly string _titleFormat = FeaturesResources.Change_accessibility_to_0;
+
+        public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var cancellationToken = context.CancellationToken;
+            var document = context.Document;
+            var syntaxFacts = document.GetLanguageService<ISyntaxFactsService>();
+            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+
+            // Innermost: We are looking for an IdentifierName. IdentifierName is sometimes at the same span as its parent (e.g. SimpleBaseTypeSyntax).
+            var diagnosticNode = root.FindNode(context.Span, getInnermostNodeForTie: true);
+            if (!syntaxFacts.IsDeclaration(diagnosticNode))
+            {
+                return;
+            }
+
+            var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            var declaredSymbol = semanticModel.GetDeclaredSymbol(diagnosticNode, cancellationToken);
+            if (declaredSymbol is not (IPropertySymbol or IMethodSymbol or IEventSymbol))
```

Comment:
are tehre tests for all these three types?   are there tests for indexers?

Reply:
The test is quite strange, because in VB `Private MustOverride` will produce `IsAbstract = false`.
Maybe this test can be removed, because the error code has specific meaning, and the implementation is not casting to them directly (casting indirectly in `GetOverridenMember`).

Reply:
sorry, that's not what i'm asking.  I'm asking if you hve actual tests that check for the error on properties, methods and events) :)